lib/miq_automation_engine/engine/miq_ae_engine.rb

```diff
@@ -67,45 +68,67 @@ def self.deliver_queue(args, options = {})
     options[:ae_state_previous] = YAML.dump(workspace.current_state_info) unless workspace.current_state_info.empty?
   end
 
+  def self.get_request_id(options)
+    # binding.pry
+    if options[:object_type].ends_with?("Request")
+      return options[:object_id]
+    end
+    if options[:object_type].ends_with?("Task")
+      return MiqRequestTask.find(options[:object_id]).miq_request_id
+    end
+    if options[:object_type].starts_with?("Service")
+      return Service.find(264).miq_request.id
+    end
+    nil
+  end
+
   def self.deliver(*args)
     options     = options_from_args(args)
     user_obj    = ae_user_object(options)
     state       = options[:state]
     vmdb_object = nil
     ae_result   = 'error'
     miq_task    = MiqTask.find(options[:open_url_task_id]) if options[:open_url_task_id]
-
+    $request_id = get_request_id(options)
+    $object_id = options[:object_id]
+    $object_type = options[:object_type]
```

Comment:
We shouldn't be using global vars here...perhaps they can be stored in the workspace?

Reply:
ok